Tasks/VsTest/task.json

```diff
@@ -1,198 +1,260 @@
+        {
+            "name": "uiTests",
             "type": "boolean",
-            "label": "Run In Parallel",
+            "label": "Test mix contains UI tests",
             "defaultValue": "false",
             "required": false,
-            "helpMarkDown": "Enable parallel execution of your tests.",
-            "groupName": "executionOptions"
+            "helpMarkDown": "*******",
+            "groupName": "testSelection"
         },
         {
             "name": "runOnlyImpactedTests",
             "type": "boolean",
-            "label": "Run Only Impacted Tests",
+            "label": "Run only impacted tests",
             "defaultValue": "False",
             "required": false,
             "helpMarkDown": "Automatically select, and run only the tests needed to validate the code change.",
-            "groupName": "tiaOptions"
+            "groupName": "testSelection",
+            "visibleRule": "testSelector = testAssemblies"
         },
         {
             "name": "runAllTestsAfterXBuilds",
             "type": "string",
-            "label": "Number Of Builds After Which To Run All Tests:",
+            "label": "Number of builds after which all tests should be run",
             "defaultValue": "50",
             "required": false,
             "helpMarkDown": "Number of builds after which to automatically run all tests. Test Impact Analysis stores the mapping between test cases and source code. It is recommended to regenerate the mapping by running all tests, on a regular basis.",
-            "groupName": "tiaOptions"
-        },
+            "groupName": "testSelection",
+            "visibleRule": "testSelector = testAssemblies && runOnlyImpactedTests = true"
+        },      
+        {  
+           "name": "testPlatform",  
+            "type": "pickList",  
+            "label": "Test platform version",  
+            "defaultValue": "14.0",  
+            "required": false,  
+            "helpMarkDown": "The version of Visual Studio test to use.",  
+            "groupName": "executionOptions",  
+            "options": {  
+                 "15.0": "Visual Studio 2017",  
+                 "14.0": "Visual Studio 2015"  
+             }  
+        },         
         {
-            "name": "vstestLocationMethod",
-            "type": "radio",
-            "label": "VSTest",
+            "name": "runSettingsFile",
+            "type": "filePath",
+            "label": "Settings file",
+            "defaultValue": "",
             "required": false,
-            "groupName": "advancedExecutionOptions",
-            "defaultValue": "version",
-            "options": {
-                "version": "Version",
-                "location": "Specify Location"
-            }
+            "helpMarkDown": "Path to runsettings file to use with the tests. Use `$(Build.SourcesDirectory)` to access the Project folder.",
+            "groupName": "executionOptions"
         },
         {
-            "name": "vsTestVersion",
```

Comment:
>vsTestVersion [](start = 21, length = 13)

if we rename this option people will always get vs 2015. specially when they are using test impact using 2017